datashuttle/utils_mod/http_utils.py

```diff
@@ -0,0 +1,66 @@
+# TODO: this is directly copy from https://github.com/brainglobe/bg-atlasapi/blob/5835887c5f9fac16ea1f2682f861411e4f789b3c/bg_atlasapi/utils.py#L132
+
+from typing import Callable
+
+import requests
+from rich.progress import (BarColumn, DownloadColumn, Progress, TextColumn,
+                           TimeRemainingColumn, TransferSpeedColumn)
+
+
+def retrieve_over_http(
```

Comment:
As mentioned in the PR, I would vendor this clearly and include the relevant license.